### PR TITLE
Add fixes to react-vite example to be able to run locally

### PR DESCRIPTION
**Pull request by josephmarkus**
### Description

- Add missing README instructions for `pnpm` installation
- Add missing `pnpm-workspace.yaml` file
- Fix TypeError: Cannot read properties of null (reading 'useState') by updating both `vite` and `@originjs/vite-plugin-federation` dependencies

### Additional context

n/a

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [x] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Code of Conduct](https://github.com/originjs/vite-plugin-federation/blob/main/CODE_OF_CONDUCT.md) and follow the [Commit Convention](https://github.com/originjs/vite-plugin-federation/blob/main/.github/commit-convention.md) guidelines.
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Comment by ruleeeer** [comment]
Glad to receive PR,but all tests failed. If the workspace is updated, please update the lock file synchronously.
image

**Comment by ruleeeer** [comment]
Also, is it convenient to explain how the error was fixed? Upgrade the vite and plugin version?

**Comment by josephmarkus** [comment]
I've just updated pnpm-lock.yaml in the root.

> is it convenient to explain how the error was fixed?

There's an explanation in commit body - https://github.com/originjs/vite-plugin-federation/pull/437/commits/fd1e9d75e3d6345cacad7c98f5d908b49e8f955c

I don't know whether you meant to add the same in pull request description?

**Comment by ruleeeer** [comment]
The CI still seems to be failing because the local `pnpm` and the version used in the test don't match?
image

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,4 +1,8 @@
-lockfileVersion: '6.0'
+lockfileVersion: '6.1'
```
Comment (ruleeeer) [inline pnpm-lock.yaml:1]: Is it because of this?

**Comment by ruleeeer** [comment]
Here seems to be the solution:https://github.com/orgs/pnpm/discussions/6633#discussioncomment-6082681

**Comment by josephmarkus** [comment]
How can I run these locally to try and see if all goes green?

**Comment by midhunadarvin** [comment]
Hi @josephmarkus 

I was able to use your branch and test out the `react-vite` example. One issue that i have seen is that the shared `react` module is being loaded twice on the host application. 

Do you know how to resolve it or any suggestions ?

It seems related to these issues :
https://github.com/originjs/vite-plugin-federation/issues/429
https://github.com/originjs/vite-plugin-federation/issues/373
https://github.com/originjs/vite-plugin-federation/issues/242

Screenshot 2023-06-21 at 8 55 13 PM

**Comment by nvhungkt** [comment]
@josephmarkus 
Have you tried this? https://github.com/originjs/vite-plugin-federation/issues/438#issuecomment-1581177095
I had the same issue but following this comment solve it.

**Comment by ruleeeer** [comment]
The start-up is normal, but running the test will fail, it is very strange, we can only temporarily remove the react-vite test cases

**Review by ruleeeer — APPROVED** [review]

**Pull request merged**